### PR TITLE
Make 1.23 the DEFAULT_RELEASE_BRANCH and add 1.23 docs 

### LINKED ISSUE #984: Create 1.23 docs

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*

closes #984 

*Description of changes:*
* Made 1.23 the `DEFAULT_RELEASE_BRANCH`
* Updated docs for 1.23, including adding docs for all previous releases of 1.23

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/contents/index.md [hunk 1]
```diff
@@ -121,6 +121,11 @@ Machine Images), but it does use the EKS Optimized AMI. See the project
 repository for the [EKS Optimized AMI](https://github.com/awslabs/amazon-eks-ami)
 if you are interested in the AL2 container runtime kernel version.
 
+#### EKS-D 1.22 Version Dependencies
```
Comment (kschumy) [inline docs/contents/index.md:124]:
```suggestion
#### EKS-D 1.23 Version Dependencies
* [v1-23-eks-4](releases/1-23/4/index.md) (August 4, 2022)
```

**Review by markapruett — APPROVED** [review]
/lgtm

**Pull request merged**